src/soilId/soilIdService.ts

```diff
@@ -17,9 +17,42 @@
 
 import { graphql } from 'terraso-client-shared/graphqlSchema';
 import { SoilIdInputData } from 'terraso-client-shared/graphqlSchema/graphql';
+import { soilDataToIdInput } from 'terraso-client-shared/soilId/soilIdFunctions';
+import { SoilData } from 'terraso-client-shared/soilId/soilIdTypes';
 import * as terrasoApi from 'terraso-client-shared/terrasoApi/api';
 import { Coords } from 'terraso-client-shared/types';
 
+export const fetchSoilMatches = async ({
+  coords,
+  siteId,
+  soilData,
+}: {
+  coords: Coords;
+  siteId?: string;
+  soilData?: SoilData;
+}) => {
+  /*
+   * Depending on the given input parameters, one of the two APIs needs to be queried.
```

Comment:
issue: please explain in more detail, i.e. when site id is present, do x, else do y because z.

Reply:
I'm struggling a bit to figure out what to comment that doesn't just paraphrase the code.

Is this what you're looking for? If not, could you help me understand what detail you're hoping for?
```
   * Depending on the given input parameters, one of the two APIs needs to be queried.
   * Sites with data use the "data based" soil matches API, as it can use soil data and location
   * Temporary locations use the "location based" soil matches API, only using location
```